cmd/manager/main.go

```diff
@@ -115,35 +139,32 @@ func main() {
 		Scheme:   mgr.GetScheme(),
 		Recorder: mgr.GetEventRecorderFor("tunnel-controller"),
 	}).SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create controller", "controller", "tunnel")
-		os.Exit(1)
+		return fmt.Errorf("unable to create tunnel controller: %w", err)
 	}
 
 	// Can query for config maps like this.
 	// For controller level configs, this may be the recommended way though https://book.kubebuilder.io/reference/markers.html
 	// We can't use this though to write config maps, and the mgr.GetClient() doesn't work because the cache isn't initialized
 	config := &v1.ConfigMap{}
 	if err := mgr.GetAPIReader().Get(ctx, types.NamespacedName{Name: configMapName, Namespace: namespace}, config); client.IgnoreNotFound(err) != nil {
-		setupLog.Error(err, fmt.Sprintf("error getting config map named '%s'", configMapName))
-		os.Exit(1)
+		return err
 	} else {
-		setupLog.Info(fmt.Sprintf("Found config map named '%s' %+v", configMapName, config))
+		setupLog.Info(fmt.Sprintf("Found config map named %q %+v", configMapName, config))
```

Comment:
total nit thing, but the `%q` fmt verb does quoting for you.

Reply:
oooh nice